src/olympia/files/helpers.py

```diff
@@ -340,6 +308,51 @@ def get_syntax(self, filename):
                 return short
         return 'plain'
 
+    def _verify_files(self, verification_list, raise_on_verify=False):
+        """Verifies that all files are properly extracted.
+
+        TODO: This should probably be extracted into a separate helper
+        once we can verify that it works as expected.
+        """
+        strip = len(self.dest)
+
+        files_to_verify = get_all_files(self.dest)
+
+        # normalize file names, strip /tmp/xxxx/
+        files = [
+            '/'.join(fname.strip('/').split('/')[2:])
+            for fname in verification_list]
+
+        difference = (
+            set([name[strip:].strip('/') for name in files_to_verify]) -
+            set(files))
+
+        if difference:
+            if raise_on_verify:
+                error_msg = (
+                    'Error verifying extraction of %s. Difference: %s' % (
+                        self.src, ', '.join(list(difference))))
+                task_log.error(error_msg)
+                raise ValueError(error_msg)
+            else:
+                task_log.warning(
+                    'Calling fsync, files from %s extraction aren\'t'
+                    ' completely available.' % self.src)
+
+                # First make sure to call fsync on the top-level directory
```

Comment:
I was asking myself the same thing, I couldn't find any references or resources as to whether or not this matters. I don't think it matters so I can move it to the end in case it does.